packages/salesforcedx-vscode-core/src/commands/forceApexTestRunCodeAction.ts

```diff
@@ -10,13 +10,51 @@ import {
   SfdxCommandBuilder
 } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
 import { nls } from '../messages';
+import { notificationService } from '../notifications';
 import {
   EmptyParametersGatherer,
   SfdxCommandlet,
   SfdxCommandletExecutor,
   SfdxWorkspaceChecker
 } from './commands';
 
+function isEmpty(value: string): boolean {
+  return !value || value.length === 0;
+}
+
+function isNotEmpty(value: string): boolean {
+  return !isEmpty(value);
+}
+
+// cache last test class and test method values to
+// enable re-running w/o command context via built-in LRU
+class ForceApexTestRunCache {
+  public lastClassTestParam: string;
+  public lastMethodTestParam: string;
+  private static instance: ForceApexTestRunCache;
+
+  public static getInstance() {
+    if (!ForceApexTestRunCache.instance) {
+      ForceApexTestRunCache.instance = new ForceApexTestRunCache();
+    }
+    return ForceApexTestRunCache.instance;
+  }
+
+  public constructor() {
+    this.lastClassTestParam = '';
+    this.lastMethodTestParam = '';
+  }
+
+  public hasCachedClassTestParam() {
+    return isNotEmpty(this.lastClassTestParam);
+  }
+
+  public hasCachedMethodTestParam() {
+    return isNotEmpty(this.lastMethodTestParam);
+  }
+}
+
```

Comment:
Could you add a `export const forceApexTestRunCacheService = ForceapexTestRunCache.getInstance()`. We would use that exported instance instead of having classes call `ForceapexTestRunCache.getInstance()`.

Again, just a matter of convention.

Reply:
Will do.